### PR TITLE
Removed Mutability Valid Values Rules - Issue1944

### PR DESCRIPTION
**Pull request by sarangan12**
Removed Mutability Valid Values Rules - https://github.com/Azure/autorest/issues/1944 

@amarzavery @vishrutshah @veronicagg @salameer Please review and approve

### CODE REVIEW

File: src/modeler/AutoRest.Swagger/Validation/Extensions/MutabilityValidValuesRule.cs [hunk 1]
```diff
@@ -1,68 +0,0 @@
...
-using AutoRest.Core.Validation;
-
-namespace AutoRest.Swagger.Validation
-{
-    public class MutabilityValidValuesRule : MutabilityExtensionRule
-    {
-        /// <summary>
-        /// Id of the Rule.
-        /// </summary>
-        public override string Id => "M2008";
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/Extensions/MutabilityValidValuesRule.cs]: Are we now missing ID and category in the "MutabilityWithReadOnlyRule" rule?

File: src/modeler/AutoRest.Swagger/Validation/Extensions/MutabilityWithReadOnlyRule.cs [hunk 2]
```diff
@@ -70,5 +71,10 @@ private bool ValidateMutabilityValuesWithReadOnlyProperty(object mutable, RuleCo
 
             return isValid;
         }
+
+        /// <summary>
+        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// </summary>
+        public override Category Severity => Category.Error;
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/Extensions/MutabilityWithReadOnlyRule.cs]: do you mind reordering code a bit, and as you're adding ID and category, place them at the beginning, with this one next, then the message template, and then the rest of the methods?

File: src/modeler/AutoRest.Swagger/Validation/Extensions/MutabilityValidValuesRule.cs (hunk 1 above)
Comment (sarangan12) [inline src/modeler/AutoRest.Swagger/Validation/Extensions/MutabilityValidValuesRule.cs]: Added ID and Category

File: src/modeler/AutoRest.Swagger/Validation/Extensions/MutabilityWithReadOnlyRule.cs (hunk 2 above)
Comment (sarangan12) [inline src/modeler/AutoRest.Swagger/Validation/Extensions/MutabilityWithReadOnlyRule.cs]: Reordered the code.

**Pull request merged**